modules/margin-liquidity-pools/src/lib.rs

```diff
@@ -167,12 +170,27 @@ decl_module! {
 			Self::deposit_event(RawEvent::SetEnabledTrades(who, pool_id, pair, enabled));
 		}
 
-		pub fn update_swap(origin, pool_id: T::LiquidityPoolId, pair: TradingPair, rate: Fixed128) {
+		pub fn set_swap_rate(origin, pair: TradingPair, long_rate: Fixed128, short_rate: Fixed128) {
+			T::UpdateOrigin::try_origin(origin)
+				.map(|_| ())
+				.or_else(ensure_root)?;
+			ensure!(long_rate >= T::MaxSwap::get().0 && long_rate <= Fixed128::zero(), Error::<T>::SwapRateTooLow);
```

Comment:
`let (min_long_rate, max_short_rate) = T::MaxSwap::get()`
@zjb0807 this will make it much easy